File: server/src/Makc2021.Layer2/Queries/List/Get/ListGetQueryOutput.cs
// Copyright (c) 2021 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

namespace Makc2021.Layer2.Queries.List.Get
{
    /// <summary>
    /// Выходные данные запроса на получение списка.
    /// </summary>
    /// <typeparam name="TItem">Тип элемента.</typeparam>
    public class ListGetQueryOutput<TItem>
    {
        #region Properties

        /// <summary>
        /// Элементы.
        /// </summary>
        public TItem[] Items { get; set; }

        /// <summary>
        /// Общее число элементов.
        /// </summary>
        public long TotalCount { get; set; }

        #endregion Properties
    }
}

File: client/makc2021-react-redux-ts/src/Layer2/Queries/List/Get/ListGetQueryOutput.ts
// Copyright (c) 2021 Maxim Kuzmin. All rights reserved. Licensed under the MIT License.

/**
 * Выходные данные запроса на получение списка элементов.
 * @template TItem Тип элемента.
 */
export interface ListGetQueryOutput<TItem> {
  /**
   * Элементы.
   */
  items: TItem;

  /**
   * Общее число элементов.
   */
  totalCount: number;
}

/**
 * Создать выходные данные запроса на получение списка элементов.
 * @template TItem Тип элемента.
 * @returns Выходные данные запроса на получение списка элементов.
 */
export function createListGetQueryOutput<TItem>(): ListGetQueryOutput<TItem> {
  return {
    totalCount: 0
  } as ListGetQueryOutput<TItem>;
}
